resources/service_systemd.rb

```diff
@@ -27,88 +27,90 @@
 property :install_path, String
 property :tomcat_user, String, default: lazy { |r| "tomcat_#{r.instance_name}" }
 property :tomcat_group, String, default: lazy { |r| "tomcat_#{r.instance_name}" }
+property :service_name, String, default: lazy { |r| "tomcat_#{r.instance_name}" }
 property :env_vars, Array, default: [
   { 'CATALINA_PID' => '$CATALINA_BASE/bin/tomcat.pid' },
 ]
 property :service_vars, Array, default: []
 
 property :service_template_source, String, default: 'init_systemd.erb'
 property :service_template_cookbook, String, default: 'tomcat'
+property :service_template_local, [true, false], default: false
 
 action :start do
-  create_init
+  action_create
 
-  service "tomcat_#{new_resource.instance_name}" do
+  service new_resource.service_name do
     provider Chef::Provider::Service::Systemd
     supports restart: true, status: true
     action :start
     only_if 'command -v java >/dev/null 2>&1 || exit 1'
   end
 end
 
+action :create do
+  template "/etc/systemd/system/#{new_resource.service_name}.service" do
```

Comment:
Any reason why you can't use the [systemd_unit](https://docs.chef.io/resources/systemd_unit/) resource instead?

Reply:
Nope; didn't realize that was a thing. I'll switch over to it.

Reply:
So the downside I see is that `systemd_unit` only expects a `content` property, is there a standard way to still utilize a template/variables to feed the creation of that?

Maybe a helper library that generates the content from a template and variables?

Reply:
You can either give it raw content or do it in a hash.  I think a helper library makes sense that generates it into a string raw content would be ideal unless it's easier to do it via a hash.

Reply:
So my thought was a helper library that behaves the same as the `template_file` resource but rather than creating a file, simply returns a string that we could feed in. We wouldn't want to lose how people may be overriding the template file in a wrapper cookbook.

It's not terribly apparent to me how to approach creating that library though (ideally reusing logic from Chef itself)

Reply:
So here's what I would do:
1. Create a method which takes arguments which relate to the variables you currently use
2. Using those arguments, create a string that is generated using those args. You might need to make smaller methods to generate the looping data.

If you want, I could write up something simple. There might also be a better way.

Reply:
Sure, feel free to write something up since I don't think I'm following.

Reply:
As we just discussed on the weekly meeting, let's keep this as-is and leave a `TODO` comment explaining it.